### ISSUE TITLE
Default Transfer Syntax for .. Image Compression is not supported

### ISSUE THREAD

**Issue by Corvin666** [fo-dicom/fo-dicom#866]
So, I've been running into this problem lately when trying to deploy a .NET Framework console app that does some work on lossy JPEG compressed xrays.

The app is built against fo-dicom 4.0.1 as a nuget package. Dicom.Native64.dll is there, app has been targeting x64 at build, works fine on dev machine and many others. I'm just copying files with the (console) app and the data to various machines at this point. I have two Win 10 x64 machines which are fresh and fully patched, it works on one (a fresh Win 10 Pro VM from Azure - didn't work before I installed 2015-2019 redist) and doesn't work on the other, (Win 10 Pro, personal laptop). Both machines have the 2013 redist and 2015-2019 redist as linked [here](https://github.com/fo-dicom/fo-dicom/wiki/Prerequisites). I tried on 4 other laptops where it works, but all of those have dev environment installed.

I'm running out of ideas here..

### Expected behavior

Call to RenderImage returns a BitmapSource

### Actual behavior
Each time RenderImage is called, an exception is thrown with the message:

_Decoding dataset with transfer syntax: JPEG Baseline (Process 1): Default Transfer Syntax for Lossy JPEG 8 bit Image Compression is not supported._

### fo-dicom version and OS/platform
Windows 10
fo-dicom 4.0.1
.NET Framework 4.7.2

**Comment by hkoestin** [comment]
When you turn on logging accordingly, do you see similar log-lines like the following?

```
[Dicom.Imaging.Codec] Searching C:\Projects\xyz\_bin\x86\Debug\Dicom.Native*.dll for Dicom codecs 
[Dicom.Imaging.Codec] Codec: JPEG Baseline (Process 1): Default Transfer Syntax for Lossy JPEG 8 Bit Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG Extended (Process 2 & 4): Default Transfer Syntax for Lossy JPEG 12 Bit Image Compression (Process 4 only) 
[Dicom.Imaging.Codec] Codec: JPEG Lossless, Non-Hierarchical (Process 14) 
[Dicom.Imaging.Codec] Codec: JPEG Lossless, Non-Hierarchical, First-Order Prediction (Process 14 [Selection Value 1]): Default Transfer Syntax for Lossless JPEG Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG 2000 Image Compression (Lossless Only) 
[Dicom.Imaging.Codec] Codec: JPEG 2000 Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG-LS Lossless Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG-LS Lossy (Near-Lossless) Image Compression 
[Dicom.Imaging.Codec] Codec: RLE Lossless 
```

Have you tried to use the Assembly Binding Log Viewer once?
You could use that to see, if binding of other assemblies maybe fails. Could be, that you are missing other assemblies to be deployed.
https://docs.microsoft.com/en-us/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer

When it is a difference between DEV machines and others, it is usually related to assemblies, which maybe are in the GAC or the like on DEV machines.

I also once had the problem, that certain Win10 machines with virus and malware detection had blocked some of the DLLs. Maybe something related to this?

![image](https://user-images.githubusercontent.com/13849875/57679093-ceb23400-762a-11e9-897e-fcb0ee271389.png)

**Comment by Corvin666** [comment]
Thanks a lot, Harald, all great suggestions, will try that.

How did you get this output? 

`[Dicom.Imaging.Codec] Searching C:\Projects\xyz\_bin\x86\Debug\Dicom.Native*.dll for Dicom codecs 
[Dicom.Imaging.Codec] Codec: JPEG Baseline (Process 1): Default Transfer Syntax for Lossy JPEG 8 Bit Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG Extended (Process 2 & 4): Default Transfer Syntax for Lossy JPEG 12 Bit Image Compression (Process 4 only) 
[Dicom.Imaging.Codec] Codec: JPEG Lossless, Non-Hierarchical (Process 14) 
[Dicom.Imaging.Codec] Codec: JPEG Lossless, Non-Hierarchical, First-Order Prediction (Process 14 [Selection Value 1]): Default Transfer Syntax for Lossless JPEG Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG 2000 Image Compression (Lossless Only) 
[Dicom.Imaging.Codec] Codec: JPEG 2000 Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG-LS Lossless Image Compression 
[Dicom.Imaging.Codec] Codec: JPEG-LS Lossy (Near-Lossless) Image Compression 
[Dicom.Imaging.Codec] Codec: RLE Lossless `

This is not assembly binder, is it?

**Comment by Corvin666** [comment]
Harald, you were spot on with the blockage of DLLs. I turned on fo-dicom logging and saw that it was reporting that it can not find the Dicom.Native*.dll despite searching in the correct folder. Checked properties window and yes, it was blocked. For some reason, all other DLLs were being loaded just fine despite having the same checkbox. I suspect there's some Windows black magic at work here. Maybe all other DLLs were being recognized as managed code and were somehow marked as safe by the executable whereas the Native64 one is treated differently either because it is being sideloaded or is not managed.

One other thing was happening here: my colleague was placing the executable inside Downloads folder which does not even show the "Unblock" checkbox so I could not unblock anything until I moved the app to another location.

For all those looking, in order to get detailed output on what is actually happening with the loading of codecs, turn on fo-dicom logging like so: `Dicom.Log.SetImplementation(Dicom.Log.ConsoleLogManager.Instance)`
Also refer to [this post](https://github.com/fo-dicom/fo-dicom/wiki/Logging-configuration).
